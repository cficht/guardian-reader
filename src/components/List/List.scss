.search-container {
  display: flex;
  justify-content: center;
  background-color: darkblue;
}

.list-container {
  ul {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    list-style-type: none;
    padding-inline-start: 0px;

    li {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      align-items: center;
      width: 20%;
      margin: 10px;
      padding: 10px;
      text-align: center;

      h3 {
        margin-block-start: 0em;
      }

      img {
        width: 100%;
      }
    }
  }
}

.pagination-container {
  display: flex;
  justify-content: center;
  background-color: darkblue;
}