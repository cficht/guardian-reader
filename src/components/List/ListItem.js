import React from 'react';
import PropTypes from 'prop-types';

export default function ListItem({ article }) {
  return (
    <li>
      <h3>{article?.webTitle}</h3>
    </li>
  );
}

ListItem.propTypes = {
  article: PropTypes.object
};
