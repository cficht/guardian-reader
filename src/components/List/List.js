import React from 'react';
import { useArticles } from '../../hooks/articleHooks';
import ListItem from './ListItem';

export default function List() {
  const { articles } = useArticles();

  const articleNodes = articles.map(article => <ListItem key={article.id} article={article}/>);

  return (
    <main>
      <ul>
        {articleNodes}
      </ul>
    </main>
  );
}
