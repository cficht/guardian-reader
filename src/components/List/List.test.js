/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';

test('renders List.js correctly', () => {
  const tree = renderer
    .create(<List />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
