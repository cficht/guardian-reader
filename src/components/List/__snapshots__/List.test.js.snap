// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List.js correctly 1`] = `
<main>
  <ul />
</main>
`;
